pkg/terra/contract_config_tracker.go

```diff
@@ -31,43 +31,51 @@ func NewContractTracker(spec OCR2Spec, jobID string, client *Client, lggr Logger
 	}
 
 	contract := Contract{
-		ChainID:     spec.ChainID,
-		JobID:       jobID,
-		ContractID:  addr,
-		terra:       client,
-		Transmitter: spec.TransmissionSigner,
-		data:        make(chan Events),
-		stop:        make(chan struct{}),
-		log:         lggr,
+		ChainID:         spec.ChainID,
+		JobID:           jobID,
+		ContractAddress: addr,
+		terra:           client,
+		Transmitter:     spec.TransmissionSigner,
+		stop:            make(chan struct{}),
+		log:             lggr,
 	}
 
-	// begin websocket subscription using terra.Subscribe
-	return &contract, contract.terra.Subscribe(context.TODO(), jobID, contract.ContractID, contract.data)
+	return &contract, nil
 }
 
 // Start creates a loop with handler for different event types
-func (ct *Contract) Start() {
-	for {
-		select {
-		case data := <-ct.data:
-			// process received data
-			for _, event := range data.Events {
-				if event == "wasm-set_config" {
-					ct.newConfig <- struct{}{}
+func (ct *Contract) Start() error {
+	// TODO:; timeout
+	txes, err := ct.terra.clientCtx.Client.Subscribe(context.TODO(), ct.JobID,
+		fmt.Sprintf("tm.event='Tx' AND execute_contract.contract_address='%s'", ct.ContractAddress))
+	if err != nil {
+		return err
+	}
+	go func() {
+		for {
+			select {
+			case data := <-txes:
+				// process received data
+				for event := range data.Events {
+					if event == "wasm-set_config" {
+						ct.newConfig <- struct{}{}
+					}
 				}
+			case <-ct.stop:
+				return
 			}
-		case <-ct.stop:
-			return
 		}
-	}
+	}()
+	return nil
 }
 
 // Close stops the event channel listener loop and unsubscribes the job from the websocket
 func (ct *Contract) Close() error {
 	// stop listening loops
 	defer close(ct.stop)
```

Comment:
`Close()` returns before all listening loops are stopped. What do you think about adding a wait group so that the close is synchronous?

Reply:
on second thought was able to just remove ws all together